README.md

```diff
@@ -1 +1,60 @@
-# ul-image-api
\ No newline at end of file
+# UL Image API
+
+The [Unified Listing](http://ul.gpii.net) is a resource for people who want to find assistive technology solutions and
+mainstream products with assistive features.  This package is used within the Unified Listing to allow:
+
+1. Contributing images and image metadata associated with [Unified Listing](http://ul.gpii.net/) entries.
+2. Reviewing and approving or rejecting contributions.
+3. Allowing users to see the status of their contributions.
+4. Viewing approved images as part of a product listing.
+
+Anyone (even users who are not logged in) can view the images and galleries associated with the "unified" source.  This
+is a curated data source created by reviewers based on imported data and user contributions.
+
+By [signing up for a free account](/api/user/signup) and [logging in](/api/user/login), users gain the ability to
+contribute updates to the data served by this API.
+
+Users can contribute new images, which are only visible to the submitter and reviewers.  An image is approved or
+rejected by a reviewer.
+
+Users can submit metadata regarding an image (for example, to suggest a better description for a given image).
+These suggestions are only visible to the submitter and reviewers.  Reviewers approve and reject contributed metadata.
+
+To contribute image data to the live Unified Listing, please visit [http://ul.gpii.net].  To try out the software
+locally, please read the "Developer Documentation" below.
+
+# Developer Documentation
+
+## The API
+
+The API provides a series of REST inputs.  For details regarding the endpoints, their accepted input, and their return
+values, see [docs/api-rest-docs.md](the API documentation).
+
+The API is built as a series of [Fluid components](http://fluidproject.org/). For more information, see
+[docs/api-component-docs.md](the API component documentation).  In addition, the API is built using the following key
+libraries:
+
+1. [Sharp](http://sharp.dimens.io/), for image resizing.
```

Comment:
The other key piece of technology is something that handles multipart/form uploads.  It needs to:

1. Accept file content encoded using multipart/form encoding.
2. Refuse to accept overly large files.
3. Refuse to read an overly large amount of field data into memory.
4. Refuse to accept unacceptable file types.

I started with the [express documentation](https://github.com/expressjs/body-parser), which mentions a range of candidates.  Of those, I almost immediately gravitated towards multer because I could see clearly from the docs that they at least intend for people to have a lot of control over the way it works from the outside.  For example, check out  [the amount of control multer gives when choosing what content to accept](https://www.npmjs.com/package/multer#limits).  You can either use declaritive options to configure it, or if that's too limiting, you can [wire in your own filter function](https://www.npmjs.com/package/multer#filefilter).  They also have clear docs on replacing the storage engine, which appealed to me as it gives us the option to explore using versioned storage down the road.

At least from their docs, all of the rest (multipart, busboy, formidable) seemed to offer only fixed configuration options.  They also do not allow you to reject files of a particular type.

In summary, multer was the only candidate that met all of our needs.  It also seems to give us the most room to expand or extend its behavior in the future.